### PR TITLE
refactor(transform): Single dataset, move conflict check up to lib.Save

### PR DESCRIPTION
**Pull request by dustmop**
Instead of a pair of read/write datasets on the starlark ds object, have only a single dataset. For checking conflicts, have Apply track which components are modified by a transform script. Compare that result against the components provided to the save path, but do so in a much more obvious way, up in lib/

### CODE REVIEW

File: transform/apply.go [hunk 1]
```diff
@@ -53,7 +54,7 @@ func NewTransformer(appCtx context.Context, loader dsref.Loader, pub event.Publi
 }
 
 // Apply applies the transform script to a target dataset
-func (t Transformer) Apply(
+func (t* Transformer) Apply(
```
Comment (b5) [inline transform/apply.go:57]: gofmt plz. Type should own the star

File: transform/startf/exec_step.go [hunk 2]
```diff
@@ -24,20 +24,20 @@ var log = golog.Logger("startf")
...
-	eventsCh  chan event.Event
-	thread    *starlark.Thread
+	starCtx    *starctx.Context
+	dsLoader   dsref.Loader
+	target     *dataset.Dataset
+	globals    starlark.StringDict
+	bodyFile   qfs.File
+	eventsCh   chan event.Event
+	thread     *starlark.Thread
+	changeList map[string]bool
```
Comment (b5) [inline transform/startf/exec_step.go:34]: again, use a `map[string]struct{}`

**Review by b5 — APPROVED** [review]
negative diffstat! Some small nits & then looks good to me

**Comment by dustmop** [comment]
Ok, now switched to using PathMap to get the components that exist in the incoming dataset. Switched the `changeSet` (previously changeList) to use full component names now, which removes the need for the `shortNameToComponent` function.

**Pull request merged**